### PR TITLE
Add "Commuter Rail Alternatives" tab to OL schedule 

### PR DESCRIPTION
**Pull request by i0sea**
#### Summary of changes
**Asana Ticket:** [Add "Commuter Rail Alternatives" tab to OL schedule ](https://app.asana.com/0/555089885850811/1202792787402845/f)

Added a commuter rail alternatives tab to the orange line page. This contains a timetable of CR stops from South Station to Forest Hills and links to the needed stops north of South Station.

Todo
- [ ] add an error check for `Schedules.Repo.by_route_ids`. 
Screen Shot 2022-08-17 at 5 17 54 PM

### CODE REVIEW

**Comment by lund5000** [comment]
Sweeeet. How much time do you think it would take to add the custom PDF links to https://cdn.mbta.com/sites/default/files/route_pdfs/2022-summer/2022-08-10-cr-hyde-park-forest-hills-ruggles-back-bay-south-station-ol-shutdown.pdf and https://cdn.mbta.com/sites/default/files/route_pdfs/2022-summer/2022-08-15-cr-oak-grove-malden-center-north-station-ol-shutdown.pdf described in the ticket? Should we do that as part of this PR or as a quick follow-up?

**Comment by lund5000** [comment]
Looking at this again, I'd also like some minor adjustments to the copy and headings:

- "at no charge" paragraph on its own at the top
- "North Side Alternatives" h2
- Haverhill link paragraph
- "South Side Alternatives" h2
- Slightly modified version of the timetable explanation: "This is a combined timetable for all commuter rail trains that will be running to south side Orange Line stations while service is suspended."
- Combined timetable and the rest of the page as it is now

dev-green mbtace com_schedules_Orange_timetable_schedule_direction destination = schedule_direction direction_id =1 schedule_direction origin = (1)

**Comment by i0sea** [comment]
> Sweeeet. How much time do you think it would take to add the custom PDF links to https://cdn.mbta.com/sites/default/files/route_pdfs/2022-summer/2022-08-10-cr-hyde-park-forest-hills-ruggles-back-bay-south-station-ol-shutdown.pdf and https://cdn.mbta.com/sites/default/files/route_pdfs/2022-summer/2022-08-15-cr-oak-grove-malden-center-north-station-ol-shutdown.pdf described in the ticket? Should we do that as part of this PR or as a quick follow-up?

@lund5000  I think this should be a quick follow up. Also think the pdfs come from the Siteweb endpoint so they'd have to put via an API I believe.

**Comment by i0sea** [comment]
Also Direction picker isn't altering the table. Will investigate and make copy changes ⌨️

File: apps/site/lib/site_web/views/helpers.ex [hunk 1]
```diff
@@ -170,7 +170,15 @@ defmodule SiteWeb.ViewHelpers do
       fa("arrow-right"),
       content_tag(:span, "to", class: "sr-only"),
       " ",
-      headsign
+      if route.id == "Orange" do
+        if direction_id == 0 do
+          "South Station"
+        else
+          "Forest Hills"
+        end
```
Comment (thecristen) [inline apps/site/lib/site_web/views/helpers.ex:178]: Are these headsigns backwards? It's making the direction picker say `Southbound -> South Station` whereas Forest Hills is south of South Station.

As a reference, on the line diagram, the southbound train goes to Forest Hills:

image

File: apps/site/lib/site_web/controllers/schedule/timetable_controller.ex [hunk 2]
```diff
@@ -109,11 +115,20 @@ defmodule SiteWeb.ScheduleController.TimetableController do
   defp all_stops(conn, _) do
     # we override the default fetch of all_stops to not use the date. We will
     # use the date to fetch the actual schedule data.
-    all_stops =
+    all_stops = if conn.assigns.route.id == "Orange" do
+
+      if conn.assigns.direction_id == 0 do
+        stop_ids = ["place-forhl","place-rugg", "place-bbsta", "place-sstat"]
```
Comment (thecristen) [inline apps/site/lib/site_web/controllers/schedule/timetable_controller.ex:121]: I think this is backwards too. The timestamps should be increasing as we go down the column:

image

File: apps/site/lib/site_web/templates/schedule/_timetable.html.eex [hunk 3]
```diff
@@ -1,4 +1,22 @@
...
+    During this shutdown, riders can simply show their CharlieCard or CharlieTicket to conductors to ride the Commuter Rail in
+    <a target="_blank" href="https://cdn.mbta.com/sites/default/files/2021-03/2021-03-23-cr-fare-zones.pdf">Zones 1A, 1 and 2 </a>at no charge.
+  </p>
+  <h2>North Side Alternatives</h2>
+  <p>
+    For service on the north side of the Orange Line, most
+    <a target="_blank" href="https://www.mbta.com/schedules/CR-Haverhill/timetable">Haverhill Line</a> trains will stop at
+    <a target="_blank" href="https://www.mbta.com/stops/place-ogmnl">Oak Grove</a> en route to
+    <a target="_blank" href="https://www.mbta.com/stops/place-mlmnl">Malden Center</a> and
+    <a target="_blank" href="https://www.mbta.com/stops/place-north">North Station</a>
```
Comment (thecristen) [inline apps/site/lib/site_web/templates/schedule/_timetable.html.eex:13]: nitpick: use the url helpers  Phoenix provides, e.g. `href="<%= stop_path(@conn, :show, "place-north") %>"`

**Review by thecristen — CHANGES_REQUESTED** [review]
🔄

**Review by lund5000 — APPROVED** [review]
I think this is good to go once the direction & order issues Cristen raised are fixed.

File: apps/site/lib/site_web/templates/schedule/_timetable.html.eex [hunk 4]
```diff
@@ -7,7 +7,7 @@
   <h2>North Side Alternatives</h2>
   <p>
     For service on the north side of the Orange Line, most
-    <a target="_blank" href="<%= stop_path(@conn, :show, "CR-Haverhill") %>">Haverhill Line</a> trains will stop at
+    <a target="_blank" href="https://www.mbta.com/schedules/CR-Haverhill/timetable">Haverhill Line</a> trains will stop at
```
Comment (lund5000) [inline apps/site/lib/site_web/templates/schedule/_timetable.html.eex:10]: Good catch! 🎉

**Review by thecristen — APPROVED** [review]

**Pull request merged**